src/msg.c

```diff
@@ -744,16 +744,18 @@ natsMsg_GetTime(natsMsg *msg)
 }
 
 natsStatus
-natsMsg_create(natsMsg **newMsg,
+natsMsg_createWithPadding(natsMsg **newMsg,
                const char *subject, int subjLen,
                const char *reply, int replyLen,
-               const char *buf, int bufLen, int hdrLen)
+               const char *buf, int bufLen, int bufPaddingSize, int hdrLen)
 {
     natsMsg     *msg      = NULL;
     char        *ptr      = NULL;
     int         bufSize   = 0;
     int         dataLen   = bufLen;
     bool        hasHdrs   = (hdrLen > 0 ? true : false);
+    // Make payload a null-terminated string and add at least one zero byte to the end
+    int         padLen    = (bufPaddingSize > 0 ? bufPaddingSize : 1);
```

Comment:
As you have seen, say the payload is 5 bytes and corresponds to `hello`, the library allocates 6 bytes and stores the `\0` at the end, so we end up with payload `hello\0`, which is convenient to get the message payload as a string (if content is a string). With your changes, the padding size incudes the `\0`, is that what you wanted? That is, suppose that for same payload, you set the option of padding size of 5, then the result would be `hello\0\0\0\0\0`, which means 5 zeros, not 5+1. It is inconsequential for normal use (if the payload is a string), and the message data length API will still return 5, but it makes the padding case be a bit different than no padding in that non padded means payload + 1, while padded means payload+padding (and not +1). If you think this is what you need for your use case, then I am fine with this change.

Reply:
Actually what was done before (adding just a single `\0` to the end) can be seen just as a special case for a generic padding implementation which is done in this PR. Now by default we have one `\0` but with the new option we can increase the number of `\0` to any arbitrary number. And yes, if I set 5 in this option then `hello\0\0\0\0\0` is exactly what I'm expecting!